### PR TITLE
Implemented Radix Sort in Kotlin

### PR DESCRIPTION
**Pull request by KartikeySharma**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Radix sort is an integer sorting algorithm that sorts data with integer keys by grouping the keys by individual digits that share the same significant position and value (place value). Radix sort uses counting sort as a subroutine to sort an array of numbers.

### Checklist

- [Tick] I've read the contribution guidelines.
- [Tick] I've checked the issue list before deciding what to submit.
- [Tick] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1815

### CODE REVIEW

**Comment by KartikeySharma** [comment]
Pls review @atarax665

**Comment by KartikeySharma** [comment]
Do let me know if any changes or updates are required @atarax665 .

File: Kotlin/sort/RadixSort/src/RadixSort.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+3
+-1
+2
+3
+4
+
+Sample Output:
+Input Array: 3 -1 2 3 4
+Array after Radix Sort is: -1 2 3 3 4
+
```
Comment (srimanikantapalakollu) [inline Kotlin/sort/RadixSort/src/RadixSort.kt]: @KartikeySharma please remove  this extra line.

File: Kotlin/sort/RadixSort/src/RadixSort.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
+fun radixSort(original: IntArray): IntArray {
```
Comment (srimanikantapalakollu) [inline Kotlin/sort/RadixSort/src/RadixSort.kt:8]: @KartikeySharma please format the code properly that should be understandable to everyone.

**Review by srimanikantapalakollu — CHANGES_REQUESTED** [review]

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 2 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt:8]: I didn't quite get you.. could you explain how is it not understandable?

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 1 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt]: Done

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 2 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt:8]: If by formatting, you mean comments then let me know.. I will add some explanation

**Comment by KartikeySharma** [comment]
@srimani-programmer Could you please review the formatting comment you added. I am not able to understand what you meant to imply

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 2 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt:8]: Added some comments explaining what the function does @srimani-programmer

**Comment by KartikeySharma** [comment]
@srimani-programmer Pls reply atleast ?

**Comment by srimanikantapalakollu** [comment]
@KartikeySharma sorry for the late reply. I'm currently out of station. So could you please contact some other mentor to review your PR. I hope you got my point.

File: Kotlin/sort/RadixSort/src/RadixSort.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// Function to Implement Radix Sort
```
Comment (atarax665) [inline Kotlin/sort/RadixSort/src/RadixSort.kt:1]: Add beginner introduction of Radix sort.

File: Kotlin/sort/RadixSort/src/RadixSort.kt [hunk 4]
```diff
@@ -0,0 +1,67 @@
+// Function to Implement Radix Sort
+// Takes an IntArray as arguments and returns a sorted IntArray to the caller function
+fun radixSort(original: IntArray): IntArray {
+    // Array needs to be mutable
+    var old = original
+    // Loop for every bit in the integers
```
Comment (atarax665) [inline Kotlin/sort/RadixSort/src/RadixSort.kt]: Remove trivial comments.

File: Kotlin/sort/RadixSort/src/RadixSort.kt [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    }
+    return old
+}
+
+fun main(args: Array<String>) {
+    print("Enter N: ");
+    val n= readLine()!!.toInt();
+    println("Enter array of N integers: ");
+    val arrays = IntArray(n) { readLine()!!.toInt() }
+    print("Input Array: ")
```
Comment (atarax665) [inline Kotlin/sort/RadixSort/src/RadixSort.kt]: We don't need to print the Input array again, you can remove this loop.

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 5 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt]: Updated. Pls review @atarax665

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 4 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt]: Updated. Pls review @atarax665

File: Kotlin/sort/RadixSort/src/RadixSort.kt (hunk 3 above)
Comment (KartikeySharma) [reply Kotlin/sort/RadixSort/src/RadixSort.kt:1]: Updated. Pls review @atarax665

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**